src/main/java/org/opentripplanner/middleware/triptracker/ManageTripTracking.java

```diff
@@ -237,4 +251,91 @@ private static <T> T getPayloadFromRequest(Request request, Class<T> payloadClas
         }
     }
 
+    /**
+     * Get the trip status by comparing the traveler's current position to an acceptable boundary around the expect
+     * position.
+     */
+    public static TripStatus getTripStatus(TrackedJourney trackedJourney, MonitoredTrip monitoredTrip) {
+        TrackingLocation lastLocation = trackedJourney.locations.get(trackedJourney.locations.size() - 1);
+        Coordinates currentPosition = new Coordinates(lastLocation.lat, lastLocation.lon);
+        ZonedDateTime currentTime = ZonedDateTime.ofInstant(lastLocation.timestamp.toInstant(), DateTimeUtils.getOtpZoneId());
+        var expectedLeg = getExpectedLeg(currentTime, monitoredTrip);
+        if (canUseLeg(expectedLeg)) {
+            List<ManageLegTraversal.Segment> segments = interpolatePoints(expectedLeg);
+            Coordinates expectedPosition = getSegmentPosition(expectedLeg.getScheduledStartTime(), currentTime, segments);
```

Comment:
Right here is the crux of the logic. Extract these functions that return the expected position:
- `getExpectedPosition(Instant instant, Itinerary itinerary)`
- `getExpectedPosition(Instant instant, Leg leg)`
 
Then you can write tests around what these function should return.

Reply:
My gut feeling is that you don't need to use a timezone because the underlying start/end time of each leg is not zoned (although `getScheduledStart/EndTime` returns zoned time info).

Reply:
Updated and working with just the start/end time for now i.e. no zoned.